honeycombio/honeycomb-opentelemetry-android
# feat(android): implement Honeycomb.configure and add smoke tests

## Which problem is this PR solving?

* Adds the `Honeycomb.configure` method, which configures OpenTelemetry for Android using `HoneycombOptions`. 
* Also adds smoke tests.

## Short description of the changes

There is a slight change to the design of `configure`. The method now returns an instance of `OpenTelemetryRum`. This is necessary because:
1. There is no global mechanism for getting an `OpenTelemetryRum`, and
2. The comments in `opentelemetry-android` disrecommend using `GlobalOpenTelemetry`.

The smoke tests are based on the ones in `honeycomb-opentelemetry-web`. The cases being tested are executed as part of the regular `androidTest` target, with their expectations implemented in bats. They are a little flaky, in that occasionally the `collector/data.json` file is empty. They are not yet included in CI, but that is likely to be in a subsequent PR.

## How to verify that this has the expected result

Follow setup instructions in `DEVELOPING.md` and then run:
```
cd smoke-tests
make smoke
make unsmoke
```


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207992777792928
